js/main.js

```diff
@@ -15,48 +14,98 @@ function getRecipeData(mealType, cusineType) {
+  $recipeName.textContent = recipeObj.name;
+  $dishDescription.appendChild($recipeName);
+
+  var $ingredientHeading = document.createElement('p');
+  $ingredientHeading.textContent = 'Ingredients';
+  $dishDescription.appendChild($ingredientHeading);
+
+  var $ingredientList = document.createElement('ul');
+  $dishDescription.appendChild($ingredientList);
+
+  for (var k = 0; k < recipeObj.ingredient.length; k++) {
+    var $ingredient = document.createElement('li');
+    $ingredient.textContent = recipeObj.ingredient[k];
+    $ingredientList.appendChild($ingredient);
+  }
+
+  var $getInstructionsLink = document.createElement('a');
+  $getInstructionsLink.textContent = 'Get Instructions';
+  $getInstructionsLink.target = '_blank';
+  $getInstructionsLink.setAttribute('class', 'get-instructions-link');
+  $getInstructionsLink.href = recipeObj.instructionsUrl;
+  $dishDescription.appendChild($getInstructionsLink);
+}
+
 function setMealAndCusineInfo(event) {
-  event.preventDefault();
-  if (event.target.getAttribute('id') === 'meal-type-form') {
-    for (var mealIndex = 0; mealIndex < $mealType.length; mealIndex++) {
-      if ($mealType[mealIndex].checked) {
-        mealType = $mealType[mealIndex].getAttribute('id');
-      }
-    }
+  if (event.target.className.includes('meal-type')) {
+    mealType = event.target.textContent;
     changeView('question-2');
-  } else if (event.target.getAttribute('id') === 'cuisine-type-form') {
-    for (var cuisineIndex = 0; cuisineIndex < $cuisineType.length; cuisineIndex++) {
-      if ($cuisineType[cuisineIndex].checked) {
-        cuisineType = $cuisineType[cuisineIndex].getAttribute('id');
-      }
-    }
+  } else if (event.target.className.includes('cuisine-type')) {
+    cuisineType = event.target.textContent;
+    changeView('recommended-dish');
   }
-  checkMealAndCusineInfo();
+  checkMealAndCusineInfo(mealType, cuisineType);
 }
 
-function checkMealAndCusineInfo(event) {
+function checkMealAndCusineInfo() {
   if (mealType && cuisineType) {
     getRecipeData(mealType, cuisineType);
   }
 }
 
 function changeView(view) {
-  if ($homePage.getAttribute('data-view') === 'home-page') {
-    $homePage.classList.add('hidden');
-    $form1.classList.remove('hidden');
-  }
-  if (view === $form2.dataset.view) {
-    $form1.classList.add('hidden');
-    $form2.classList.remove('hidden');
+  for (var i = 0; i < $pageContainer.length; i++) {
+    if ($pageContainer[i].dataset.view === view) {
+      $pageContainer[i].classList.remove('hidden');
+    } else {
+      $pageContainer[i].classList.add('hidden');
+    }
+
   }
 }
 
-$mealTypeForm.addEventListener('submit', setMealAndCusineInfo);
-$cuisineTypeForm.addEventListener('submit', setMealAndCusineInfo);
-$startQuizButton.addEventListener('click', changeView);
+function getDataValue() {
+  var dataView = event.target.getAttribute('data-view');
+  changeView(dataView);
+}
+
+function reloadPage() {
+  location.reload();
+}
```

Comment:
😱 😱 😱 😱 😱 

You should _never_ use `location.reload()`. It entirely refreshes your page, which is amongst the most resource-heavy actions you can take. It is essentially the same as if you decided you wanted to move your couch to the other side of the room, but instead of just moving it, you blow up your entire house and rebuild it with the couch on the other side of the room :joy: